src/main/java/com/bandwidth/sqs/publisher/MessagePublisher.java

```diff
@@ -0,0 +1,12 @@
+package com.bandwidth.sqs.publisher;
+
+import com.amazonaws.services.sqs.model.SendMessageResult;
+
+import java.time.Duration;
+import java.util.Optional;
+
+import io.reactivex.Single;
+
+public interface MessagePublisher<T> {
+    Single<SendMessageResult> publishMessage(T newMessage, String newQueueUrl, Optional<Duration> delay);
```

Comment:
hm, `Optional<Duration>` makes is kinda clear, but so would 
```java
publishMessage(aMessage, aQueue, Duration.ofMillis(0))
```
or even a convenience method
```java
default Single<SendMessageResult> publishMessage(T newMessage, String newQueueUrl) {
  publishMessage(newMessage, newQueue, Duration.ofMillis(0));
}
```